Assets/MixedRealityToolkit/Inspectors/Profiles/MixedRealityToolkitConfigurationProfileInspector.cs

```diff
@@ -117,6 +117,7 @@ public override void OnInspectorGUI()
                 {
                     ScriptableObject profile = CreateInstance(nameof(MixedRealityToolkitConfigurationProfile));
                     var newProfile = profile.CreateAsset("Assets/MixedRealityToolkit.Generated/CustomProfiles") as MixedRealityToolkitConfigurationProfile;
+                    UnityEditor.Undo.RecordObject(MixedRealityToolkit.Instance, "Create new profiles");
```

Comment:
I think actually putting this in the `CreateAsset` overload would be a little better as we'd then get undo capabilities for each created profile as well.

Reply:
My intent was to cover the assignment to the ActiveProfile field, so I think the Undo.RecordObject needs to be here in the outer function.

Reply:
@StephenHodgson raises an interesting point about undo and creation of the asset itself. I was just interested in the side-effect of the undo, which is that it tells Unity the the object is dirty so that it can be saved. But if you actually want to be able to fully undo the whole shebang then I suppose you'd also want to delete the asset that's created. Perhaps that could be accomplished via Undo.RegisterCreatedObjectUndo inside the AssetDatabase.CreateAsset extension method, but I'm not sure if that works with assets and what all the consequences of that could be. I see several calls in MRTK to CreateAsset that don't seem to be creating an Undo record and I'm hesitant to wade into that territory.

For now, I'd like to stick with this fix as-is to ensure that the change is saved as described in #4109. @StephenHodgson, if you like you could open a new issue to suggest that calls to AssetDatabase.CreateAsset should be keeping Undo records. But I'm not sure if that's sufficiently important.

Reply:
Resolving the conversation. (Please let me know if that's not the etiquette for PR creator to resolve.)

Reply:
I think that's fine :) Your reply makes sense to me.

I'm not sure it warrants an new issue, as I think it's just a nice to have feature.

Thanks @ForrestTrepte!